bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setTagsProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        TagManager tagManager = this.resourceResolver.adaptTo(TagManager.class);
+        JSONArray tags = (JSONArray) json.get(key);
+        ArrayList<Tag> tagList = new ArrayList<>();
+
+        for (int i = 0; i < tags.length(); i++) {
+            Tag tag = tagManager.resolve(tags.getString(i));
+            if (tag == null) {
+                LOG.warn("Tag '{}' could not be found. Skipping tag for node '{}'.", tags.getString(i), node.getPath());
+                continue;
+            }
+
+            tagList.add(tag);
+        }
+
+        if (tagList.size() > 0) {
+            Resource parentResource = this.resourceResolver.getResource(node.getPath());
+            tagManager.setTags(parentResource, tagList.toArray(new Tag[0]));
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Tags added for node '{}'.", node.getPath());
+            }
+        }
+    }
+
+    /**
+     * Set generic array property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setArrayProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        JSONArray rawValues = (JSONArray) json.get(key);
+        Object firstVal = rawValues.get(0);
+        Value[] propertyValues = new Value[rawValues.length()];
+
+        if (firstVal instanceof Boolean) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getBoolean(i));
+            }
+        } else if (firstVal instanceof Double) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getDouble(i));
+            }
+        } else if (firstVal instanceof Integer) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getInt(i));
+            }
+        } else if (firstVal instanceof Long) {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getLong(i));
+            }
+        } else {
+            for (int i = 0; i < rawValues.length(); i++) {
+                propertyValues[i] = this.valueFactory.createValue(rawValues.getString(i));
+            }
+        }
+
+        node.setProperty(key, propertyValues);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Array property '{}' added for node '{}'.", key, node.getPath());
```

Comment:
I wonder if setting each individual `property` (with maybe the exception of complex things like mixins) should be at the `trace` level.